src/napari_vedo_bridge/_cutter_widget.py

```diff
@@ -1,120 +1,190 @@
+from vedo.utils import is_ragged
 
 
 class VedoCutter(QWidget):
 
     def __init__(self, napari_viewer=None):
         super().__init__()
 
+        self.mesh = None
+
+        self.cutter_widget = None
+
+        self.mesh_color = "yellow6"
+        self.mesh_backcolor = "purple7"
+        self.mesh_lighting = "shiny"
+    
+        self.vedo_message = Text2D(font='Calco', c='white')
+        self.vedo_axes = None
+
         self.napari_viewer = napari_viewer
         uic.loadUi(os.path.join(Path(__file__).parent, "./vedo_extension.ui"), self)
 
-    #     Qt.QMainWindow.__init__(self, parent)
-    #     self.frame = Qt.QFrame()
-        # self.layout = Qt.QVBoxLayout()
         self.vtkWidget = QVTKRenderWindowInteractor()
         self.layout().insertWidget(0, self.vtkWidget)
-        self.plt = Plotter(qt_widget=self.vtkWidget, bg='blackboard', axes=0)
-        self.plt.show(interactive=False)
 
         self.pushButton_send_back.clicked.connect(self.send_to_napari)
         self.pushButton_get_from_napari.clicked.connect(self.get_from_napari)
 
         self.pushButton_box_cutter.clicked.connect(self.box_cutter_tool)
         self.pushButton_sphere_cutter.clicked.connect(self.sphere_cutter_tool)
         self.pushButton_plane_cutter.clicked.connect(self.plane_cutter_tool)
+        self.pushButton_load_mesh.clicked.connect(self._load_mesh)
+
+        self.plt = Plotter(qt_widget=self.vtkWidget, bg='bb', interactive=False)
+        self.plt += self.vedo_message
+        self.plt += Text2D(
+            "vedo "+_vedo_version,
+            pos='top-right',
+            font='Calco',
+            c='k5',
+            s=0.5,
+        )
+        self.plt.show()
 
-        self.plane_cutter_widget = None
-        self.box_cutter_widget = None
-        self.sphere_cutter_widget = None
 
     def get_from_napari(self):
         """
         Get the currently selected layer from napari and display it in vedo
-        """
+        """        
+        if self.cutter_widget:
+            self.plt.remove(self.cutter_widget)
+        self.plt.remove(self.mesh, self.vedo_axes)
 
         self.currently_selected_layer = self.napari_viewer.layers.selection.active
-        self.mesh = Mesh(self.currently_selected_layer.data[:2])  # only vertices and faces
-        #mesh.pointdata = self.currently_selected_layer.features  # scalars
-        # Create renderer and add the vedo objects and callbacks
+        if self.currently_selected_layer:
+            points = self.currently_selected_layer.data[0].astype(float)
+            faces = self.currently_selected_layer.data[1].astype(int)
+            scalars = self.currently_selected_layer.data[2]
+            self.mesh = Mesh([points, faces])  # only vertices and faces       
+            if len(scalars) > 0:
+                self.mesh.pointdata['scalars'] = scalars
+        else:
+            # TEST mesh
+            self.mesh = Mesh(dataurl+"mouse_brain.stl")
```

Comment:
Rather than building it in here we could also provide the vedo example data library as separate functions?

Reply:
You mean the test mesh?